src/synapse/azext_synapse/_params.py

```diff
@@ -3,159 +3,204 @@
 def load_arguments(self, _):
     # synapse spark
-
-    for scope in ['batch', 'session', 'session-statement']:
+    for scope in ['job', 'session', 'statement']:
         with self.argument_context('synapse spark ' + scope) as c:
             c.argument('workspace_name', help='The name of the workspace.')
-            c.argument('spark_pool_name', help='The name of the spark pool.')
-
-    for scope in ['synapse spark batch', 'synapse spark session']:
-        with self.argument_context(scope + ' create') as c:
-            c.argument('job_name', arg_type=name_type, help='The spark batch or session job name.')
-            c.argument('file', help='The URI of file.')
-            c.argument('class_name', help='The class name.')
-            c.argument('args', nargs='+', help='The arguments of the job.')
-            c.argument('jars', nargs='+', help='The array of jar files.')
-            c.argument('files', nargs='+', help='The array of files URI.')
-            c.argument('archives', nargs='+', help='The array of archives.')
-            c.argument('conf', help='The configuration of spark batch job.')
-            c.argument('driver_memory', help='The memory of driver.')
-            c.argument('driver_cores', help='The number of cores in driver.')
-            c.argument('executor_memory', help='The memory of executor.')
-            c.argument('executor_cores', help='The number of cores in each executor.')
-            c.argument('num_executors', help='The number of executors.')
-            c.argument('tags', arg_type=tags_type)
-            c.argument('detailed', action='store_true',
-                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+            c.argument('spark_pool_name', help='The name of the Spark pool.')
 
+    for scope in ['synapse spark job', 'synapse spark session']:
         with self.argument_context(scope + ' list') as c:
             c.argument('from_index', help='Optional parameter specifying which index the list should begin from.')
-            c.argument('detailed', action='store_true',
-                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
             c.argument('size',
                        help='The size of the returned list.By default it is 20 and that is the maximum.')
 
-    with self.argument_context('synapse spark batch create') as c:
-        c.argument('language', arg_type=get_enum_type(['SCALA', 'PYTHON', 'DOTNET'], default='SCALA'),
-                   help='The spark batch job language.')
+    with self.argument_context('synapse spark job submit') as c:
+        c.argument('main_definition_file', help='The main file used for the job.')
+        c.argument('main_class_name',
+                   help='The fully-qualified identifier or the main class that is in the main definition file.')
+        c.argument('command_line_arguments', nargs='+',

```

Comment:
what's this? seems the name is too general for job submit command

Reply:
This parameter means the arguments that is needed when run job in command line.


---
In reply to: [424947356](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424947356) [](ancestors = 424947356)